### DISCUSSION TITLE
Managing DHCP Leases

### DISCUSSION THREAD

**Discussion by ajtatum** [O-X-L/ansible-opnsense#44] [Q&A] [answered]
I apologize if this is a dumb question. I'm just starting to learn Ansible and the world of automation possibilities there are out there.

One somewhat tedious task for me is whenever I create a new LXC or VM in Proxmox, I like to assign it an IP address with a Proxmox MAC address and the host name.

I'm not entirely sure if this is possible, but in my head I'm imagining that I could create a playbook where I supply the host name and it would generate a random MAC address for whatever I'm creating and then pass those variables into OPNsense and once it's finished it would go back to Proxmox and build the LXC/VM.

I hope that makes sense.

Thanks,
AJ

**Comment by ansibleguy** [comment]
Greetings!

This should be doable WHEN OPNSense supports managing the DHCP-leases using an API.

Currently this is not yet possible. So we cannot implement this functionality as Ansible module.

I created a feature-request at the OPNSense repo - https://github.com/opnsense/core/issues/7108

Also related to: https://github.com/ansibleguy/collection_opnsense/issues/35 and https://github.com/ansibleguy/collection_opnsense/discussions/14#discussioncomment-5678082

**Comment by ansibleguy** [comment]
It seems the new/full DHCP API should become available in the next few months. After that I will implement it as Ansible modules.

**Comment by ansibleguy** [comment, marked as answer]
You can follow/watch this feature-request https://github.com/ansibleguy/collection_opnsense/issues/45 for updates on this subject

**Comment by ajtatum** [comment]
Awesome! Thank you!
